src/components/ui/Badge/Badge.tsx

```diff
@@ -1,26 +1,49 @@
 import { Badge as UIBadge } from '@faststore/ui'
 import type { ReactNode } from 'react'
 import React from 'react'
+import IconButton from 'src/components/ui/IconButton'
+import Icon from 'src/components/ui/Icon'
 
 import './badge.scss'
 
-export type BadgeVariants = 'info' | 'highlighted' | 'neutral'
+export type BadgeVariants = 'info' | 'highlighted' | 'neutral' | 'interactive'
```

Comment:
Hmm, I think the `interactive Badge` shouldn't be a new variant. Since the `Badge` can be `info`/`highlighted`/`neutral` and `interactive` at the same time. Maybe we can deal with the `interactive` checking the `onClose` callback existence so that we can deal with both _states_? What do you think?

Reply:
Yeah, you are right. I miss the point of using the action together with the other cases

Reply:
Agreed!

I was thinking about how we could apply [the needed different styles when in interactive mode](https://github.com/vtex-sites/base.store/pull/396/files#diff-a1eda55d5a5721215d3eed0d9797b4df0781a3c9c53d51b9c7c76f5b5521f980R39) assuming `data-store-badge-variant=interactive` won't be possible. I think we can either:

1. Put the interactive information in another data attribute (e.g. `data-store-badge-mode=interactive`).
2. Join the variant **and** interactive information in the same data attribute (e.g. `data-store-badge-mode="info interactive"`) and match the interactive in CSS via the `[attr*=value]` selector (e.g. `data-store-badge-variant*=interactive`).

> <a href="https://developer.mozilla.org/en-US/docs/Web/CSS/Attribute_selectors#:~:text=followed)%20by%20value.-,%5Battr*%3Dvalue%5D,-Represents%20elements%20with">[attr*=value]</a>
> Represents elements with an attribute name of attr whose value contains at least one occurrence of value within the string.

Reply:
I agreed with Filipe's suggestion. It seems a nice approach for this scenario.